hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/HoodieDeltaStreamer.java

```diff
@@ -635,7 +637,7 @@ public DeltaSync getDeltaSync() {
                     HoodieTimeline.COMPACTION_ACTION, scheduledCompactionInstantAndRDD.get().getLeft().get()));
                 asyncCompactService.get().waitTillPendingAsyncServiceInstantsReducesTo(cfg.maxPendingCompactions);
               }
-              if (cfg.isAsyncClusteringEnabled()) {
+              if (HoodieClusteringConfig.from(props).isAsyncClusteringEnabled()) {
```

Comment:
this does a full load each time?

Reply:
Can we retain the resolved clustering config above as a member?

Reply:
Can you elaborate what you refer to by "load"? All of the properties are already in `props`

Reply:
We certainly can cache this object though